addons/account/account.py

```diff
@@ -278,6 +286,8 @@ def write(self, vals):
             if ('company_id' in vals and journal.company_id.id != vals['company_id']):
                 if self.env['account.move'].search([('journal_id', 'in', self.ids)], limit=1):
                     raise UserError(_('This journal already contains items, therefore you cannot modify its company.'))
```

Comment:
remove this code, not needed anymore because of this: https://github.com/odoo/odoo/pull/7077/files#r34915488